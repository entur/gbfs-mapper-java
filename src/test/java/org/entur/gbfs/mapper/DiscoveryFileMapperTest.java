package org.entur.gbfs.mapper;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.io.IOException;
import java.net.URL;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@ExtendWith({SnapshotExtension.class})
class DiscoveryFileMapperTest {
    private static final ObjectMapper objectMapper = new ObjectMapper();

    private Expect expect;

    @SnapshotName("gbfs_v2_3_to_v3_0_discovery_file_snapshot")
    @Test
    void testMapDiscoveryFile() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v2_3/gbfs.json");
        org.mobilitydata.gbfs.v2_3.gbfs.GBFS testSubject = objectMapper.readValue(resource, org.mobilitydata.gbfs.v2_3.gbfs.GBFS.class);
        org.mobilitydata.gbfs.v3_0.gbfs.GBFSGbfs mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }

    @SnapshotName("gbfs_v3_0_to_v2_3_discovery_file_snapshot")
    @Test
    void testMapDiscoveryFileInverse() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v3_0/gbfs.json");
        org.mobilitydata.gbfs.v3_0.gbfs.GBFSGbfs testSubject = objectMapper.readValue(resource, org.mobilitydata.gbfs.v3_0.gbfs.GBFSGbfs.class);
        org.mobilitydata.gbfs.v2_3.gbfs.GBFS mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }
}
