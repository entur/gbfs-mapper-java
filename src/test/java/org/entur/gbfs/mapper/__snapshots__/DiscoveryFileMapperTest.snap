gbfs_v2_3_to_v3_0_discovery_file_snapshot=[
  {
    "last_updated" : "1970-01-19T14:18:47.710+00:00",
    "ttl" : 300,
    "version" : "3.0-RC2",
    "data" : {
      "feeds" : [
        {
          "name" : "gbfs_versions",
          "url" : "https://test.com/gbfs_versions"
        },
        {
          "name" : "system_information",
          "url" : "https://test.com/system_information"
        },
        {
          "name" : "vehicle_types",
          "url" : "https://test.com/vehicle_types"
        },
        {
          "name" : "station_information",
          "url" : "https://test.com/station_information"
        },
        {
          "name" : "station_status",
          "url" : "https://test.com/station_status"
        },
        {
          "name" : "vehicle_status",
          "url" : "https://test.com/free_bike_status"
        },
        {
          "name" : "system_regions",
          "url" : "https://test.com/system_regions"
        },
        {
          "name" : "system_pricing_plans",
          "url" : "https://test.com/system_pricing_plans"
        },
        {
          "name" : "system_alerts",
          "url" : "https://test.com/system_alerts"
        },
        {
          "name" : "geofencing_zones",
          "url" : "https://test.com/geofencing_zones"
        }
      ]
    }
  }
]


gbfs_v3_0_to_v2_3_discovery_file_snapshot=[
  {
    "last_updated" : 1606727710,
    "ttl" : 300,
    "version" : "2.3",
    "data" : {
      "en" : {
        "feeds" : [
          {
            "name" : "gbfs_versions",
            "url" : "https://test.com/gbfs_versions"
          },
          {
            "name" : "system_information",
            "url" : "https://test.com/system_information"
          },
          {
            "name" : "vehicle_types",
            "url" : "https://test.com/vehicle_types"
          },
          {
            "name" : "station_information",
            "url" : "https://test.com/station_information"
          },
          {
            "name" : "station_status",
            "url" : "https://test.com/station_status"
          },
          {
            "name" : "free_bike_status",
            "url" : "https://test.com/free_bike_status"
          },
          {
            "name" : "system_regions",
            "url" : "https://test.com/system_regions"
          },
          {
            "name" : "system_pricing_plans",
            "url" : "https://test.com/system_pricing_plans"
          },
          {
            "name" : "system_alerts",
            "url" : "https://test.com/system_alerts"
          },
          {
            "name" : "geofencing_zones",
            "url" : "https://test.com/geofencing_zones"
          }
        ]
      }
    }
  }
]